src/metabase/pulse/render/body.clj

```diff
@@ -695,6 +695,20 @@
       [:img {:style (style/style {:display :block :width :100%})
              :src   (:image-src image-bundle)}]]}))
 
+(s/defmethod render :gauge :- common/RenderedPulseCard
+  [_chart-type render-type _timezone-id :- (s/maybe s/Str) card _dashcard data]
+  (let [image-bundle (image-bundle/make-image-bundle
+                      render-type
+                      (js-svg/gauge card data))]
+    {:attachments
+     (when image-bundle
+       (image-bundle/image-bundle->attachment image-bundle))
+
+     :content
+     [:div
+      [:img {:style (style/style {:display :block :width :100%})
+             :src   (:image-src image-bundle)}]]}))
+
```

Comment:
Maybe this is over-optimizing, but I'm wondering if the `{:attachments (...) :content [...]}` bit should be extracted out into a helper function. Gets repeated all over the place.

Reply:
Yeah, I feel that too. I'll try to do something about it.